### ISSUE TITLE
[apex] ApexCRUDViolation: SObject name case sensitive in Schema checks

### ISSUE THREAD

**Issue by AndrewStopchenko-SO** [pmd/pmd#5633]
**Affects PMD Version:** 
7.11.0

**Rule:**
[ApexCRUDViolation](https://pmd.github.io/latest/pmd_rules_apex_security.html#apexcrudviolation)

**Description:**
ApexCRUDViolation rule is case sensitive for matching the case/capitalization of the SObject name that gets checked in the `Schema.*` checks and the case/capitalization of the SOQL query/DML being tested.

Also, for Schema checks it only recognizes _**"sObjectType"**_ capitalization specifically (with first letter lowercase) for `Schema.sObjectType.<sObjectApiName>.<furtherChecks>` checks, which is slightly misleading - because e.g. VSCode IntelliSense suggests `Schema.SObjectType` as the Type's name (first letter uppercase, as usual for class names).

_This one might be opinionated though, since SF docs are also using the `Schema.sObjectType` capitalization in the [code samples](https://developer.salesforce.com/docs/atlas.en-us.apexref.meta/apexref/apex_class_Schema_SObjectType.htm) here and there..._

**Code Sample demonstrating the issue:**

```apex
// false-positive violations reported
public with sharing class Sample1 {

    public static void exampleSchemaChecksCaseSensitive1() {
        // Notice "contact" is lowercase in the Schema checks
        if (Schema.sObjectType.contact.isAccessible()) {
            SObject[] recs = [SELECT Id FROM Contact LIMIT 1];    // false-positive violation
        }
    }

    public static void exampleSchemaChecksCaseSensitive2() {
        if (Schema.sObjectType.Lead.isAccessible()) {
            // Notice "lead" is lowercase in the SOQL query
            SObject[] recs = [SELECT Id FROM lead LIMIT 1];   // false-positive violation
        }
    }

    public static void exampleSchemaChecksSObjectTypeCaseSensitive() {
        // Notice that "SObjectType" has capital "S" (which is the actual Apex Type name)
        if (Schema.SObjectType.Account.isAccessible()) {
            SObject[] recs = [SELECT Id FROM Account LIMIT 1];    // false-positive violation
        }
    }
}

// ========================================
// ========================================

// resolving false-positive violations above
public with sharing class Sample2 {

    public static void exampleSchemaChecksCaseSensitive1Fixed() {
        // Fixed by capitalizing Contact in Schema checks
        if (Schema.sObjectType.Contact.isAccessible()) {
            SObject[] recs = [SELECT Id, Name FROM Contact LIMIT 1];
        }
    }

    public static void exampleSchemaChecksCaseSensitive2Fixed() {
        if (Schema.sObjectType.lead.isAccessible()) {
            // Fixed by lowercasing Lead object name in SOQL query (to match the casing in Schema checks)
            SObject[] recs = [SELECT Id, Name FROM lead LIMIT 1];
        }
    }

    public static void exampleSchemaChecksSObjectTypeCaseSensitiveFixed() {
        // Fixed by using exactly the "sObjectType" capitalization in Schema checks
        if (Schema.sObjectType.Account.isAccessible()) {
            SObject[] recs = [SELECT Id, Name FROM Account LIMIT 1];
        }
    }
}
```

This problem becomes even harder to spot for CustomObjects that have multiple words in their name, where the API name becomes e.g. `My_Custom_Rule__c`. Spotting some casing difference in Schema checks might become even harder then:
```apex
// API name matches
if (Schema.My_Custom_Rule__c.isAccessible()) {
  ...
}

// API name typed in manually with "rule" lowercased - causes false-positive, but really hard to spot in the code
if (Schema.My_Custom_rule__c.isAccessible()) {
  ...
}
```

**Expected outcome:**
Expected the CustomObject's API name to have case-insensitive checks between sObject API name's casing in the tested SOQL/DML and the Schema checks.

**Running PMD through:** *[CLI | [ApexPMD](https://github.com/ChuckJonas/vscode-apex-pmd) VSCode extension]*
